jslack-api-client/src/main/java/com/github/seratch/jslack/api/methods/SlackApiException.java

```diff
@@ -19,8 +19,26 @@ public SlackApiException(Response response, String responseBody) {
     }
 
     public SlackApiException(SlackConfig config, Response response, String responseBody) {
+        this(response, responseBody, parse(config, responseBody));
+    }
+
+    public SlackApiException(Response response, String responseBody, SlackApiErrorResponse error) {
+        super(buildErrorMessage(response, error));
         this.response = response;
         this.responseBody = responseBody;
+        this.error = error;
+    }
+
+    private static String buildErrorMessage(Response response, SlackApiErrorResponse error) {
+        String message = "status: " + response.code();
+        if (error != null) {
+            return message + ", error: " + error.getError() + ", needed: " + error.getNeeded() + ", warning: " + error.getProvided() + ", warning: " + error.getWarning();
```

Comment:
This has the text `warning` twice, i think first one should be `provided`?

Would it be easier to just make this line instead `return message + error.toString()` as SlackApiErrorResponse already has toString() implemented from the lombok `@Data` annotation?

Reply:
Good catch, thanks! >warning

>Would it be easier to just make this line instead return message + error.toString() as SlackApiErrorResponse already has toString() implemented from the lombok @Data annotation?

True, it would be much easier but I don't think it's good as the string would be much longer and it would be like "SlackApiErrorResponse(....)".

I don't think it's a good way